jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdAuthenticator.java

```diff
@@ -381,17 +401,29 @@ public Authentication validateRequest(ServletRequest req, ServletResponse res, b
         if (uri == null)
             uri = URIUtil.SLASH;
 
-        mandatory |= isJSecurityCheck(uri);
-        if (!mandatory)
-            return new DeferredAuthentication(this);
+        HttpSession session = request.getSession(false);
+        if (_openIdConfiguration.isRespectIdTokenExpiry() && hasExpiredIdToken(session))
+        {
+            logoutWithoutRedirect(request);
```

Comment:
add comment to say what will happen... i.e they will be logged out and the fall through to code below that will offer a challenge.  Hence they will be redirected to the login service rather than the configured logout page. right???

Reply:
I refactored this section to make it clearer. Take a look and let me know if you think this is better.